pkg/pillar/dpcreconciler/linux.go

```diff
@@ -687,6 +707,81 @@ func (r *LinuxDpcReconciler) clusterStatusChanged(
 			newStatus.ClusterIPPrefix)
 }
 
+// kubeUserServicesChanged checks if the Kubernetes user services configuration has changed
+func (r *LinuxDpcReconciler) kubeUserServicesChanged(
+	newServices types.KubeUserServices) bool {
```

Comment:
I would suggest using method `generics.EqualSets` (or `generics.EqualSetsFn`) for `KubeUserServices.UserService` and `KubeUserServices.UserIngress` instead of building these temporary maps.

Reply:
Ok, will do.